algorithms/src/main/kotlin/io/kinference/algorithms/gec/corrector/GECCorrector.kt

```diff
@@ -0,0 +1,191 @@
+package io.kinference.algorithms.gec.corrector
+
+import io.kinference.algorithms.gec.postprocessing.GecPostprocessor
+import io.kinference.algorithms.gec.preprocessing.*
+import io.kinference.algorithms.gec.utils.*
+import kotlin.math.min
+
+/**
+ * data class for Correction result from each iteration
+ */
+data class CorrectionResult(val sentence: String, val corrections: SentenceCorrections)
+
+/**
+ * Main class for correction generation.
+ * @param model - Seq2Logits class which can predict tags for each token in sequence
+ * @param textProcessor - TransformersTextprocessor class for processing sequences
+ * @param labelsVocab - label vocabulary
+ * @param dTagsVocab - detection tags vocabulary
+ * @param verbsVocab - verbs form vocabulary
+ * @param preprocessor - sequence preprocessor
+ * @param postprocessor - correction postprocessor
+ * @param iterations - quantity of iterations which model should do for returning corrections
+ * @param minCorrectionProb - minimum probability for correction
+ * @param minErrorProb - minimum probability for error
+ * @param confidence - bias for not-changing strategy
+ *
+ */
+class GECCorrector(val model: Seq2Logits,
+                   private val textProcessor: TransformersTextprocessor,
+                   labelsVocab: Vocabulary,
+                   dTagsVocab: Vocabulary,
+                   private val verbsVocab: VerbsFormVocabulary,
+                   private val preprocessor: GecPreprocessor,
+                   private val postprocessor: GecPostprocessor,
+                   val iterations: Int = 3,
+                   minCorrectionProb: Double = 0.0,
+                   minErrorProb: Double = 0.0,
+                   confidence: Double = 0.0) {
+    private val tagger = GecTagger(
+        model = model,
+        textProcessor = textProcessor,
+        labelsVocabulary = labelsVocab,
+        dTagsVocabulary = dTagsVocab,
+        minCorrectionProb = minCorrectionProb,
+        minErrorProb = minErrorProb,
+        confidence = confidence
+    )
+
+    /**
+     * calculate correction for batch sentence for evaluation
+     */
+    fun toCorrectedSentences(sentences: List<String>): List<CorrectionResult> {
```

Comment:
It is probably better to call it `correct`. Since `toCorrectedSentences` sounds like it is conversion, not actual processing.